driver: wifi: eswifi: Fix how we get SPI bus

galak opened the pull request:
We were hard coding the SPI bus name to "SPI_3".  Instead we should use
ESWIFI0_BUS_NAME as its generated from the device tree.

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

galak committed:
driver: wifi: eswifi: Fix how we get SPI bus

We were hard coding the SPI bus name to "SPI_3".  Instead we should use
DT_INVENTEK_ESWIFI_ESWIFI0_BUS_NAME as its generated from the device
tree.

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

nashif commented on drivers/wifi/eswifi/eswifi_bus_spi.c:
```diff
@@ -202,7 +202,7 @@ int eswifi_spi_init(struct eswifi_dev *eswifi)
 
 	/* SPI DEV */
-	spi->spi_dev = device_get_binding("SPI_3");
+	spi->spi_dev = device_get_binding(ESWIFI0_BUS_NAME);
```
where is ESWIFI0_BUS_NAME coming from?

galak replied:
Generated from dts.

nashif replied:
shouldn't this be DT_something?

galak replied:
In theory, but we also generate a set of defines based on "aliases" so we have:

```
#define ESWIFI0_BASE_ADDRESS                                                            DT_ST_STM32_SPI_FIFO_40003C00_INVENTEK_ESWIFI_1_BASE_ADDRESS
#define ESWIFI0_BUS_NAME                                                                DT_ST_STM32_SPI_FIFO_40003C00_INVENTEK_ESWIFI_1_BUS_NAME
...
```

I'll push a fix to clean that up as part of this PR.

galak committed:
driver: wifi: eswifi: Use DT_ prefixed names

The driver was using the straight alias generated defines, however we
want to use DT_ prefixed defines so its clear that the values are coming
from DT.

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>
